test/e2e/handler/error_messages_formatting_test.go

```diff
@@ -0,0 +1,150 @@
+}
+
+var _ = Describe("NodeNetworkState", func() {
+	var (
+		defaultPolicy = "test-policy"
+
+		messagesToRemove = [...]string{
+			"DEBUG    Async action: Create checkpoint started",
+			"DEBUG    Checkpoint None created for all devices",
+			"Traceback (most recent call last):",
+			"DEBUG    Nispor: current network state",
+			"WARNING  libnm version",
+			"rolling back desired state configuration: failed running probes after network changes: ",
+			"failed running probe 'ping' with after network reconfiguration -> currentState:",
+			"warnings.warn",
+		}
+	)
+
+	Context("with invalid field", func() {
+		var (
+			messagesToKeep = [...]string{
+				"libnmstate.error.NmstateVerificationError",
+				"desired",
+				"current",
+				"difference",
+			}
+		)
+
+		BeforeEach(func() {
+			createPolicyAndWaitForEnactmentCondition(defaultPolicy, createInterfaceWithInvalidField, nodes[0])
+		})
+
+		It("should discard disarranged parts of the message", func() {
+			for _, failMessage := range messagesToRemove {
+				Expect(enactmentConditionsStatus(nodes[0], defaultPolicy).Find(nmstate.NodeNetworkConfigurationEnactmentConditionFailing).Message).NotTo(ContainSubstring(failMessage))
+			}
+		})
+
+		It("should keep desired parts of the message", func() {
+			for _, failMessage := range messagesToKeep {
+				Expect(enactmentConditionsStatus(nodes[0], defaultPolicy).Find(nmstate.NodeNetworkConfigurationEnactmentConditionFailing).Message).To(ContainSubstring(failMessage))
+			}
+		})
+	})
+
+	Context("with invalid interface name", func() {
+		var (
+			messagesToKeep = [...]string{
+				"libnmstate.error.NmstateLibnmError",
+				"No suitable device found for this connection",
+				"mismatching interface name",
+			}
+		)
+
+		BeforeEach(func() {
+			createPolicyAndWaitForEnactmentCondition(defaultPolicy, createInterfaceWithInvalidName, nodes[0])
+		})
+
+		It("should discard disarranged parts of the message", func() {
+			for _, failMessage := range messagesToRemove {
+				Expect(enactmentConditionsStatus(nodes[0], defaultPolicy).Find(nmstate.NodeNetworkConfigurationEnactmentConditionFailing).Message).NotTo(ContainSubstring(failMessage))
+			}
+		})
+
+		It("should keep desired parts of the message", func() {
+			for _, validMessage := range messagesToKeep {
+				Expect(enactmentConditionsStatus(nodes[0], defaultPolicy).Find(nmstate.NodeNetworkConfigurationEnactmentConditionFailing).Message).To(ContainSubstring(validMessage))
+			}
+		})
+	})
+
+	Context("with ping fail", func() {
+		var (
+			messagesToKeep = [...]string{
+				"failed to retrieve default gw at runProbes",
+			}
+		)
+
+		BeforeEach(func() {
+			createPolicyAndWaitForEnactmentCondition(defaultPolicy, createInterfaceWithPingFail, nodes[0])
+		})
+
+		AfterEach(func() {
+			resetDesiredStateForNodes()
+			By("Remove the policy")
+			deletePolicy("test-policy")
+		})
+
+		It("should discard disarranged parts of the message", func() {
+
+			for _, failMessage := range messagesToRemove {
```

Comment:
maybe unwantedMessage/desiredMessage would be a better name for these variables?